components/servo/site_data_manager.rs

```diff
@@ -2,22 +2,97 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
+use bitflags::bitflags;
 use net_traits::ResourceThreads;
+use rustc_hash::FxHashMap;
+use storage_traits::StorageThreads;
+use storage_traits::webstorage_thread::{OriginDescriptor, StorageType};
+
+bitflags! {
+    #[derive(Clone, Copy, Debug, PartialEq)]
+    pub struct StorageTypes: u32 {
+        const LOCAL_STORAGE   = 1 << 0;
+        const SESSION_STORAGE = 1 << 1;
+
+        const ALL =
+                   Self::LOCAL_STORAGE.bits() |
+                   Self::SESSION_STORAGE.bits();
+    }
```

Comment:
`bitflags` already generates a `fn all()`, so this is unnecessary. See https://docs.rs/bitflags/latest/bitflags/example_generated/struct.Flags.html#method.all.